src/enc.cc

```diff
@@ -1231,11 +1232,12 @@ void Encoder::SinglePassScan() {
   const int mb_y_max = H_ / block_h_;
   const QuantizeBlockFunc quantize_block = use_trellis_ ? TrellisQuantizeBlock
                                                         : quantize_block_;
+  const bool have_coeffs = have_coeffs_;
   for (int mb_y = 0; mb_y < mb_h_; ++mb_y) {
     const bool yclip = (mb_y == mb_y_max);
     for (int mb_x = 0; mb_x < mb_w_; ++mb_x) {
       if (!CheckBuffers()) return;
-      if (!have_coeffs_) {
+      if (!have_coeffs) {
         in = in_blocks_;
         GetSamples(mb_x, mb_y, yclip | (mb_x == mb_x_max), in);
         fDCT_(in, mcu_blocks_);
```

Comment:
x86 can call using a pointer argument, so no gain there. Arm `blr`s from a register, but given the register pressure it would just add to the stack usage.

Reply:
agreed